options/rtdl/generic/main.cpp

```diff
@@ -814,6 +814,53 @@ void __dlapi_enter(uintptr_t *entry_stack) {
 #endif
 }
 
+#if __MLIBC_GLIBC_OPTION
+
+extern "C" [[gnu::visibility("default")]] int _dl_find_object(void *address, dl_find_object *result) {
+	for(size_t i = 0; i < initialRepository->loadedObjects.size(); i++) {
```

Comment:
Since the iterator variable is not used it would be more idiomatic to use a range based for loop here.